falcon/typing.py

```diff
@@ -15,25 +15,40 @@
 
 from __future__ import annotations
 
+from enum import auto
+from enum import Enum
 import http
 from typing import (
     Any,
+    Awaitable,
     Callable,
     Dict,
     List,
+    Literal,
+    Optional,
     Pattern,
     Protocol,
     Tuple,
     TYPE_CHECKING,
+    TypeVar,
     Union,
 )
 
 if TYPE_CHECKING:
     from falcon import asgi
+    from falcon.asgi_spec import AsgiEvent
     from falcon.request import Request
     from falcon.response import Response
 
 
+class _Missing(Enum):
+    MISSING = auto()
+
+
+_T = TypeVar('_T')
+MISSING = _Missing.MISSING
```

Comment:
Could we underscore this constant too then, if we are using it outside of this module?

Reply:
my idea was to cleanup all the typing aliases at the same time in a final PR once all things are typed